tests/integration/pyramid_tests.py

```diff
@@ -284,3 +288,144 @@ def test_local_trace_in_context(self):
         child_span = self.server_observer.on_child_span_created.call_args[0][0]
         context, server_span = self.observer.on_server_span_created.call_args[0]
         self.assertNotEqual(child_span.context, context)
+
+
+class ConfiguratorHeaderTrustTests(unittest.TestCase):
+    def setUp(self):
+        configurator = Configurator()
+        configurator.add_route("example", "/example", request_method="GET")
+        configurator.add_route("trace_context", "/trace_context", request_method="GET")
+
+        configurator.add_view(
+            example_application, route_name="example", renderer="json")
+
+        configurator.add_view(
+            local_tracing_within_context, route_name="trace_context", renderer="json")
+
+        configurator.add_view(
+            render_exception_view,
+            context=ControlFlowException,
+            renderer="json",
+        )
+
+        configurator.add_view(
+            render_bad_exception_view,
+            context=ControlFlowException2,
+            renderer="json",
+        )
+
+        mock_filewatcher = mock.Mock(spec=FileWatcher)
+        mock_filewatcher.get_data.return_value = {
+            "secrets": {
+                "secret/trace/hmac": {
+                    "type": "versioned",
+                    "current": "asdfasdfasdfasdf",
+                },
+            },
+            "vault": {
+                "token": "test",
+                "url": "http://vault.example.com:8200/",
+            }
+        }
+        secrets = store.SecretsStore("/secrets")
+        secrets._filewatcher = mock_filewatcher
+
+        self.observer = mock.Mock(spec=BaseplateObserver)
+        self.server_observer = mock.Mock(spec=ServerSpanObserver)
+        def _register_mock(context, server_span):
+            server_span.register(self.server_observer)
+        self.observer.on_server_span_created.side_effect = _register_mock
+
+        self.baseplate = Baseplate()
+        self.baseplate.register(self.observer)
+        self.baseplate_configurator = BaseplateConfigurator(
+            self.baseplate,
+            edge_context_factory=EdgeRequestContextFactory(secrets),
+            secret_store=secrets
+        )
+        configurator.include(self.baseplate_configurator.includeme)
+        self.context_init_event_subscriber = mock.Mock()
+        configurator.add_subscriber(self.context_init_event_subscriber, ServerSpanInitialized)
+        app = configurator.make_wsgi_app()
+        self.test_app = webtest.TestApp(app)
+
+    def test_hmac_valid(self):
+        self.test_app.get("/trace_context", headers={
+            "X-Trace": "1234",
```

Comment:
Let's try to use realistic values here.

> Trace identifiers are 64 or 128-bit, but all span identifiers within a trace are 64-bit. All identifiers are opaque.

[link](https://github.com/openzipkin/b3-propagation#identifiers)

(There's good chance I've violated this in the past. I'm sorry.)

Reply:
0000000000000001234 is a viable 64 bit number :)